src/Algebra/Graph/Undirected.hs

```diff
@@ -698,6 +700,24 @@ toSymmetricRelation :: Ord a => Graph a -> SR.Relation a
 toSymmetricRelation = foldg SR.empty SR.vertex SR.overlay SR.connect
 {-# INLINE toSymmetricRelation #-}
 
+-- | Complement of a graph.
+-- Complexity: /O(m^2+n)/ time, /O(m+n)/ memory where
```

Comment:
This still seems incorrect. The complexity is O(n^2) both in terms of time and memory, isn't it?

Reply:
Do you mean `O(m^2)`? my understanding was that `m` was the edge count, so by creating a clique we're increasing the number of edges while the number of vertices is constant.

Reply:
@sphaso No, I do mean O(n^2).

If we start with a graph having n = 100 vertices and, say, m = 1000 edges, the `cliqueG` will contain ~100^2 edges, not ~1000^2 edges! After all, a graph on 100 vertices can't have more than ~100^2 edges :-)

Interestingly, O(m^2) is not even an upper bound! For if you start with a graph having n = 100 vertices and m = 0 edges, the `cliqueG` will still contain ~100^2 edges, no ~0^2 edges :-)

Reply:
Digging deeper into the code: while O(n^2) is the right memory complexity, the time complexity might turn out to be O(n^2 * m) if the list difference `\\` operation is quadratic.

Reply:
@snowleopard makes sense, sorry for the many mistakes in this part.